tests/test_dagman_configurator.py

```diff
@@ -0,0 +1,131 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Unit tests for DagmanConfigurator class."""
+
+import logging
+import os
+import tempfile
+import unittest
+from pathlib import Path
+
+from lsst.ctrl.bps import BpsConfig
+from lsst.ctrl.bps.htcondor import HTCDag
+from lsst.ctrl.bps.htcondor.dagman_configurator import DagmanConfigurator
+
+logger = logging.getLogger("lsst.ctrl.bps.htcondor")
+
+
+class DagmanConfiguratorTestCase(unittest.TestCase):
+    """Unit tests for DagmanConfigurator class."""
+
+    def setUp(self):
+        self.config = BpsConfig(
+            {
+                "site": {
+                    "foo": {
+                        "wmsConfig": {"DAGMAN_USE_STRICT": 1},
+                    },
+                },
+                "wmsConfig": {"DAGMAN_USE_STRICT": 0},
+            }
+        )
+
+    def tearDown(self):
+        pass
+
+    def testInitDefaultSearchOptions(self):
+        """Test object instantiation with default search options."""
+        configurator = DagmanConfigurator(self.config)
+        self.assertIn("DAGMAN_USE_STRICT", configurator.options)
+        self.assertEqual(configurator.options["DAGMAN_USE_STRICT"], 0)
+        self.assertIsNone(configurator.config_path)
+        self.assertIsNone(configurator.prefix)
+
+    def testInitCustomSearchOptions(self):
+        """Test object instantiation with custom search options."""
+        configurator = DagmanConfigurator(self.config, search_opts={"curvals": {"computeSite": "foo"}})
+        self.assertIn("DAGMAN_USE_STRICT", configurator.options)
+        self.assertEqual(configurator.options["DAGMAN_USE_STRICT"], 1)
+        self.assertIsNone(configurator.config_path)
+        self.assertIsNone(configurator.prefix)
+
+    def testInitUnsupportedDagmanOption(self):
+        """Test object instantiation with unsupported DAGMAN options."""
+        self.config[".wmsConfig.DAGMAN_UNSUPPORTED_OPTION"] = "foo"
+        with self.assertLogs(logger=logger, level="WARNING") as cm:
+            DagmanConfigurator(self.config)
+        self.assertIn("DAGMAN_UNSUPPORTED_OPTION", cm.output[0])
+
+    def testInitNoWmsConfig(self):
+        """Test object instantiation fails when no WMS-specific options."""
+        del self.config["wmsConfig"]
+        with self.assertRaisesRegex(KeyError, "not found"):
+            DagmanConfigurator(self.config)
+
+    def testPrepare(self):
+        """Test if the method creates the configuration file."""
+        configurator = DagmanConfigurator(self.config)
+        with tempfile.TemporaryDirectory(ignore_cleanup_errors=True) as tmpdir:
+            configurator.prepare("dagman.conf", prefix=tmpdir)
+            self.assertIn(Path(tmpdir), list(configurator.config_path.parents))
+            self.assertTrue(configurator.config_path.is_file())
+            self.assertTrue(configurator.config_path.read_text(), "DAGMAN_USE_STRICT = 0")
```

Comment:
Test that the "invalid" wmsConfig options aren't written as well as any without values?